pkg/storage/unified/resource/server.go

```diff
@@ -329,6 +370,37 @@ func (s *server) Init(ctx context.Context) error {
 	return s.initErr
 }
 
+func (s *server) getClientToDistributeRequest(namespace string) *RingClient {
+	ringHasher := fnv.New32a()
+	_, err := ringHasher.Write([]byte(namespace))
+	if err != nil {
+		s.log.Error("Error hashing namespace. Will not distribute request", "err", err)
+		return nil
+	}
+
+	rs, err := s.distributor.Ring.Get(ringHasher.Sum32(), ring.NewOp([]ring.InstanceState{ring.ACTIVE}, func(s ring.InstanceState) bool {
```

Comment:
I would suggest extracting `ring.NewOp([]ring.InstanceState{ring.ACTIVE}, func(s ring.InstanceState) bool { ... })` into a variable and reuse it. No need to recreate it on every call.